test/e2e/case38_install_operator_test.go

```diff
@@ -1,126 +1,518 @@
+			incorrectOpGroupYAML = "../resources/case38_operator_install/incorrect-operator-group.yaml"
+			incorrectOpGroupName = "incorrect-operator-group"
+			scopedOpGroupYAML    = "../resources/case38_operator_install/scoped-operator-group.yaml"
+			scopedOpGroupName    = "scoped-operator-group"
+		)
 
-		Context("When there is no OperatorGroup spec specified", Ordered, func() {
-			BeforeAll(func() {
-				// Create ns op-2
-				utils.Kubectl("create", "ns", case38OpPolicyDefaultOgNS)
-				// Apply a policy in ns op-2
-				utils.Kubectl("apply", "-f", case38OpPolicyDefaultOg, "-n", case38OpPolicyDefaultOgNS)
-				// Check if applied properly
-				op := utils.GetWithTimeout(clientManagedDynamic, gvrOperatorPolicy, case38OpPolicyDefaultOgName,
-					case38OpPolicyDefaultOgNS, true, defaultTimeoutSeconds)
-				Expect(op).NotTo(BeNil())
+		BeforeAll(func() {
+			utils.Kubectl("create", "ns", opPolTestNS)
+			DeferCleanup(func() {
+				utils.Kubectl("delete", "ns", opPolTestNS)
 			})
 
-			It("Should create a default OperatorGroup", func() {
-				og := utils.GetWithTimeout(clientManagedDynamic, gvrOperatorGroup,
-					case38DefaultOgName, case38OpPolicyDefaultOgNS, true, defaultTimeoutSeconds)
+			utils.Kubectl("apply", "-f", incorrectOpGroupYAML, "-n", opPolTestNS)
 
-				Expect(og).NotTo(BeNil())
-			})
+			createObjWithParent(parentPolicyYAML, parentPolicyName,
+				opPolYAML, opPolTestNS, gvrPolicy, gvrOperatorPolicy)
+		})
 
-			It("Should create the Subscription from the spec", func() {
-				sub := utils.GetWithTimeout(clientManagedDynamic, gvrSubscription,
-					case38SubscriptionName, case38OpPolicyDefaultOgNS, true, defaultTimeoutSeconds)
+		It("Should initially be NonCompliant", func() {
+			check(
+				opPolName,
+				true,
+				[]policyv1.RelatedObject{{
+					Object: policyv1.ObjectResource{
+						Kind:       "OperatorGroup",
+						APIVersion: "operators.coreos.com/v1",
+						Metadata: policyv1.ObjectMetadata{
+							Name:      incorrectOpGroupName,
+							Namespace: opPolTestNS,
+						},
+					},
+					Compliant: "NonCompliant",
+					Reason:    "Resource found but does not match",
+				}, {
+					Object: policyv1.ObjectResource{
+						Kind:       "OperatorGroup",
+						APIVersion: "operators.coreos.com/v1",
+						Metadata: policyv1.ObjectMetadata{
+							Name:      scopedOpGroupName,
+							Namespace: opPolTestNS,
+						},
+					},
+					Compliant: "NonCompliant",
+					Reason:    "Resource not found but should exist",
+				}},
+				metav1.Condition{
+					Type:    "OperatorGroupCompliant",
+					Status:  metav1.ConditionFalse,
+					Reason:  "OperatorGroupMismatch",
+					Message: "the OperatorGroup found on the cluster does not match the policy",
+				},
+				"the OperatorGroup found on the cluster does not match the policy",
+			)
+		})
+		It("Should match when the OperatorGroup is manually corrected", func() {
+			utils.Kubectl("delete", "operatorgroup", incorrectOpGroupName, "-n", opPolTestNS)
```

Comment:
OperatorGroup is specified and another OperatorGroup also exists, then what happens?

Reply:
Can we add one more case(it)